CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs

```diff
@@ -49,16 +49,36 @@ public override async Task<bool> CanProvideCredentialsAsync(Uri uri)
                 return false;
             }
 
-            var validHosts = EnvUtil.GetHostsFromEnvironment(Logger, EnvUtil.SupportedHostsEnvVar, new[]
+             var validHosts = EnvUtil.GetHostsFromEnvironment(Logger, EnvUtil.SupportedHostsEnvVar, new[]
+             {
+                 ".pkgs.vsts.me", // DevFabric
+                 ".pkgs.codedev.ms", // DevFabric
+                 ".pkgs.codeapp.ms", // AppFabric
+                 ".pkgs.visualstudio.com", // Prod
+                 ".pkgs.dev.azure.com" // Prod
+             });
+
+            bool isValidHost = validHosts.Any(host => uri.Host.EndsWith(host, StringComparison.OrdinalIgnoreCase));
+            if (isValidHost)
             {
-                ".pkgs.vsts.me", // DevFabric
-                ".pkgs.codedev.ms", // DevFabric
-                ".pkgs.codeapp.ms", // AppFabric
-                ".pkgs.visualstudio.com", // Prod
-                ".pkgs.dev.azure.com" // Prod
-            });
-
-            return validHosts.Any(host => uri.Host.EndsWith(host, StringComparison.OrdinalIgnoreCase)) || await authUtil.IsVstsUriAsync(uri);
+                Verbose(string.Format(Resources.HostAccepted, uri.Host));
+                return true;
+            }
+
+            var feedUriSource = await authUtil.GetFeedUriSource(uri);
+            if (feedUriSource == IFeedUriSource.Hosted)
+            {
+                Verbose(Resources.ValidHeaders);
+                return true;
+            }
+
+            if (feedUriSource == IFeedUriSource.OnPrem)
+            {
+                Verbose(Resources.OnPremDetected);
+                return false;
+            }
+
```

Comment:
Might want to add logging here for the `External` case, e.g. "Feed is not Azure DevOps."